### PR TITLE
Feature checks an arrangement against a roster

### PR DESCRIPTION
**Pull request by deonjo**
If anyone is missing, it will display a label next to the person as missing. If the person is present in both, it will label that person as present.

Screenshot 2019-08-05 21 14 35

### CODE REVIEW

File: src/containers/rosterChecker/rosterChecker.jsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                readOnly: true,
+                renderer: this.renderValidity
+            }
+        ];
+    }
+
+    generateCurrentRoster = () => {
+        const itemList = []
+        console.log(this.props.real)
+        if (!isEmpty(this.props.real)) {
```
Comment (crystaldou) [inline src/containers/rosterChecker/rosterChecker.jsx]: why is this called real? what is 'real'?

File: src/containers/rosterChecker/rosterChecker.jsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    }
+
+    processChangeItemName = (row, columnTitle, previous, current) => {
+        const newRosterClone = cloneDeep(this.state.newRoster);
+        const validityOfItem = this.checkIfNameInRoster(current, this.state.currentRoster) ? PRESENT : MISSING;
+        newRosterClone[row] = {name: current, validity: validityOfItem}
+        this.setState({...this.state, newRoster: newRosterClone})
+    }
+
+    processChange = (row, columnTitle, previous, current) => {
```
Comment (crystaldou) [inline src/containers/rosterChecker/rosterChecker.jsx]: what is prev/current?

File: src/containers/rosterChecker/rosterChecker.jsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        const itemList = []
+        console.log(this.props.real)
+        if (!isEmpty(this.props.real)) {
+            console.log("generating item list...");
+        
+            this.props.real.items.forEach(item => {
+                const validityOfItem = this.checkIfNameInRoster(item.name, this.state.newRoster) ? PRESENT : MISSING;
+                itemList.push({name: item.name, validity: validityOfItem});
+            });
+            this.setState({...this.state, currentRoster: itemList});
```
Comment (crystaldou) [inline src/containers/rosterChecker/rosterChecker.jsx]: i don't think you need `..this.state` here / other places

File: src/containers/rosterChecker/rosterChecker.jsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+            td.style.backgroundColor = SHAMROCK_GREEN;
+        }
+        else if (value === MISSING) {
+            td.style.backgroundColor = CINNABAR_RED;
+        }
+        return td.innerHTML = value;
+    }
+
+    checkIfNameInRoster = (name, roster) => {
+        return roster.filter(row => {
```
Comment (crystaldou) [inline src/containers/rosterChecker/rosterChecker.jsx]: i think you can use `includes`

File: src/containers/rosterChecker/rosterChecker.jsx (hunk 1 above)
Comment (deonjo) [reply src/containers/rosterChecker/rosterChecker.jsx]: jeff chiu named it as such

File: src/containers/rosterChecker/rosterChecker.jsx (hunk 2 above)
Comment (deonjo) [reply src/containers/rosterChecker/rosterChecker.jsx]: previous is the previous entry of the cell, current is the current entry of the cell.

File: src/containers/rosterChecker/rosterChecker.jsx (hunk 4 above)
Comment (deonjo) [reply src/containers/rosterChecker/rosterChecker.jsx]: i think includes doesn't take in a predicate

**Pull request merged**